src/patternfly/components/Form/examples/Form.md

```diff
@@ -600,3 +617,4 @@ cssPrefix: pf-c-form
 | `.pf-m-inline` | `.pf-c-form__group-control` | Modifies form group children to be inline (this is primarily for radio buttons and checkboxes). |
 | `.pf-m-stack` | `.pf-c-form__group-control` | Modifies form group children to be stacked with space between children. |
 | `.pf-m-expanded` | `.pf-c-form__field-group` | Modifies an expandable field group for the expanded state. |
+| `.pf-m-grid-sm`, .pf-m-grid-md`, `.pf-m-grid-lg`, `.pf-m-grid-xl`, `.pf-m-grid-2xl` | `.pf-c-form` | Changes horizontal layout to vertical, at suffixed breakpoint. By default, pf-c-global--breakpoint--md is used |
```

Comment:
It looks to me like you're adding support for modifying when the horizontal layout is applied. Typically we would do that via `.pf-m-horizontal{-on-[breakpoint]}`.

Seems like you could do it with something like this:

```
// Add this to top of form.scss - creates a map of the breakpoint suffixes to use
$pf-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "lg", "xl", "2xl");

// updates the mixin name to something more unique to avoid namespace conflicts
@mixin pf-c-form--m-horizontal {

    // default horizontal styles applied at medium breakpoint
    @media screen and (min-width: $pf-global--breakpoint--md) {
      @content;
    }

    // custom xs style since this isn't in our maps - maps are based on mobile first design. This may not even be a valid option based on design but I guess it doesn't hurt to add.
    &-on-xs {
        @media screen and (min-width: $pf-global--breakpoint--xs) {
          @content;
        }
    }

    // loop to build the rest of the breakpoints
    @each $breakpoint, $breakpoint-value in $pf-c-form--m-horizontal--breakpoint-map {
        &-on-#{$breakpoint} {
            @include pf-apply-breakpoint($breakpoint) {
               @content
            }
        }
    }
}

// you could place `@include pf-c-form--m-horizontal {}` in the existing `&.pf-m-horizontal` block like this:
 .pf-c-form {
     &.pf-m-horizontal {
        @include pf-c-form--m-horizontal {
            /* grid styles */
        }
     }
 }
```